hadoop-jobs/src/main/java/com/linkedin/hadoop/jobs/HdfsWaitJob.java

```
@@ -0,0 +1,155 @@
+   * It splits the string up using whitespace as the delimiter. Then each time unit
+   * and corresponding integer is converted to its millisecond representation. The total
+   * millisecond value is then returned.
+   *
+   * @param prop The string value of either freshness, timeout or sleepTime
+   * @throws NumberFormatException
+   * @return The amount of milliseconds prop corresponds to as a long
+   */
+  public long parseTime(String prop) throws NumberFormatException {
+    long totalTime = 0;
+    String[] strArray = prop.split("\\s+");
+
+    for (int i = 0; i < strArray.length; i++) {
+      long time = Long.valueOf(strArray[i].substring(0, strArray[i].length() - 1));
+      char unit = strArray[i].charAt(strArray[i].length() - 1);
+
+      if (unit == 'S') {
+        totalTime += TimeUnit.SECONDS.toMillis(time);
+      } else if (unit == 'M') {
+        totalTime += TimeUnit.MINUTES.toMillis(time);
+      } else if (unit == 'H') {
+        totalTime += TimeUnit.HOURS.toMillis(time);
+      } else if (unit == 'D') {
+        totalTime += TimeUnit.DAYS.toMillis(time);
+      } else {
+        log.info("ERROR: Invalid input for -> " + prop + ". JOB TERMINATED.");
+        throw new NumberFormatException();
+      }
+    }
+    return totalTime;
+  }
+
+  /**
+   * Method checkDirectory loops through the folders pointed to by dirPath, and will
+   * cause the job to succeed if any of the folders are fresh enough.
+   *
+   * @param dirPath The path to the directory we are searching for fresh folders
+   * @param freshness The timeframe in which the folder has to have been modified by
+   * @throws IOException
+   * @return A boolean value corresponding to whether a fresh folder was found
+   */
+  public boolean checkDirectory(String dirPath, long freshness) throws IOException {
+    FileSystem fileSys = FileSystem.get(getConf());
+    FileStatus[] status = (fileSys == null) ? null : fileSys.listStatus(new Path(dirPath));
```

Ok to drop the (fileSys == null) check here and let that throw a NullPointerException if the FileSystem.get call returns null